.github/actions/update-releases-json/index.js

```diff
@@ -0,0 +1,240 @@
+const actionUtils = require('../action-utils.js');
+const path = require('path');
+
+async function run() {
+    const [core, github] = await actionUtils.installAndRequirePackages("@actions/core", "@actions/github");
+
+    const releasesDataFile = core.getInput("releases_json_file", { required: true });
+    let octokit = undefined;
+    {
+        const auth_token = core.getInput("auth_token", { required: false });
+        if (auth_token !== undefined && auth_token.length > 0) {
+            octokit = github.getOctokit(auth_token);
+        }
+    }
+
+    const endOfSupportDiscussionCategory = core.getInput("end_of_support_discussion_category", { required: false });
+    const supportedFrameworks = core.getInput("supported_frameworks", { required: false });
+
+    const repoOwner = github.context.payload.repository.owner.login;
+    const repoName = github.context.payload.repository.name;
+
+    const releasePayload = github.context.payload.release;
+
+    try {
+        const releasesData = JSON.parse(await actionUtils.readFile(releasesDataFile));
+
+        if (releasePayload !== undefined) {
+            const deprecatedRelease = addNewReleaseVersion(releasePayload, supportedFrameworks, releasesData);
+            if (deprecatedRelease !== undefined && octokit !== undefined && endOfSupportDiscussionCategory !== undefined && endOfSupportDiscussionCategory.length > 0) {
```

Comment:
Should this write a warning if octokit is undefined so that we know it didn't try to create a discussion announcement? Is octokit realistically ever going to be undefined?